xbmc/video/VideoDatabase.cpp

```diff
@@ -2874,13 +2888,30 @@ void CVideoDatabase::ClearBookMarksOfFile(int idFile, CBookmark::EType type /*=
   }
 }
 
+void CVideoDatabase::GetEpisodeByBookmarkID(int bookmarkID, long& season, long& episode)
```

Comment:
You need to add the usual protections like the try-catch block and checking if the call to `query()` actually worked.